src/sig_stfl/sig_stfl.h

```diff
@@ -469,105 +483,118 @@ OQS_API void OQS_SIG_STFL_free(OQS_SIG_STFL *sig);
 
 /**
- * Frees an OQS_SIG_STFL_SECRET_KEY object that was constructed by OQS_SECRET_KEY_new.
+ * Free an OQS_SIG_STFL_SECRET_KEY object that was constructed by OQS_SECRET_KEY_new.
  *
  * @param[in] sig The OQS_SIG_STFL_SECRET_KEY object to free.
  * @return OQS_SUCCESS if successful, or OQS_ERROR if the object cannot be freed.
  */
 OQS_API void OQS_SIG_STFL_SECRET_KEY_free(OQS_SIG_STFL_SECRET_KEY *sk);
 
 /**
- * OQS_SIG_STFL_SECRET_KEY_SET_lock .
+ * Attach a locking mechanism to a secret key object.
  *
- * Sets function to prevent multiple processes from using the sk at the same time.
+ * This allows for proper synchronization in a multi-threaded or multi-process environment,
+ * by ensuring that a secret key is not used concurrently by multiple entities, which could otherwise lead to security issues.
  *
- * @param[in] sk secret key pointer to be updated
- * @param[in] lock function pointer
+ * @param[in] sk Pointer to the secret key object whose lock function is to be set.
+ * @param[in] lock Function pointer to the locking routine provided by the application.
+ * @return None.
  *
  */
 OQS_API void OQS_SIG_STFL_SECRET_KEY_SET_lock(OQS_SIG_STFL_SECRET_KEY *sk, lock_key lock);
 
 /**
- * OQS_SIG_STFL_SECRET_KEY_SET_unlock .
+ * Attach an unlock mechanism to a secret key object.
  *
- * Sets function to prevent multiple processes from using the sk at the same time.
+ * This allows for proper synchronization * in a multi-threaded or multi-process environment,
+ * by ensuring that a secret key is not used concurrently by multiple entities, which could otherwise lead to security issues.
  *
- * @param[in] sk secret key pointer to be updated
- * @param[in] unlock function pointer
+ * @param[in] sk Pointer to the secret key object whose unlock function is to be set.
+ * @param[in] unlock Function pointer to the unlock routine provided by the application.
+ * @return None.
  *
  */
 OQS_API void OQS_SIG_STFL_SECRET_KEY_SET_unlock(OQS_SIG_STFL_SECRET_KEY *sk, unlock_key unlock);
 
 /**
- * OQS_SIG_STFL_SECRET_KEY_SET_mutex .
+ * Assign a mutex function to handle concurrency control over the secret key.
  *
- * Sets function to prevent multiple processes from using the sk at the same time.
+ * This is to ensure that only one process can access or modify the key at any given time.
  *
- * @param[in] sk secret key pointer to be updated
- * @param[in] mutex function pointer
+ * @param[in] sk A pointer to the secret key that the mutex functionality will protect.
+ * @param[in] mutex A function pointer to the desired concurrency control mechanism.
+ * @return None.
  *
  */
 OQS_API void OQS_SIG_STFL_SECRET_KEY_SET_mutex(OQS_SIG_STFL_SECRET_KEY *sk, void *mutex);
 
 /**
- * OQS_SIG_STFL_SECRET_KEY_lock .
- *
- * Locks the secret key so only one application that holds the lock can access it.
+ * Lock the secret key to ensure exclusive access in a concurrent environment.
```

Comment:
I'd assume this function will fail if no mutex has been set?! Worthwhile stating? When does this function need to be called?

Reply:
Caught the case when no mutex is set in https://github.com/open-quantum-safe/liboqs/pull/1655/commits/efccbe58a339266838a0617df8562886edc068da.  
Updated in https://github.com/open-quantum-safe/liboqs/pull/1655/commits/c577c2ed8d6597c1766f5d9612397ed876061aa3